src/Authenticator/CookieAuthenticator.php

```diff
@@ -154,7 +157,21 @@ protected function _createPlainToken($identity): string
         $usernameField = $this->getConfig('fields.username');
         $passwordField = $this->getConfig('fields.password');
 
-        return $identity[$usernameField] . $identity[$passwordField];
+        $salt = '';
+
+        if ($this->getConfig('salt')) {
+            if (is_string($this->getConfig('salt'))) {
+                $salt = $this->getConfig('salt');
+            } else {
+                $salt = Security::getSalt();
+            }
+
+            if (!is_string($salt) || strlen($salt) === 0) {
+                throw new InvalidArgumentException('Salt must be a non-empty string.');
+            }
+        }
```

Comment:
```suggestion
        $salt = $this->getConfig('salt', '');
        if ($salt === true) {
            $salt = Security::getSalt();
        } elseif ($salt === false) {
            $salt = '';
        } elseif (!is_string($salt)) {
            throw new InvalidArgumentException('Salt must be a non-empty string.');
        }
```

Reply:
With this, when the application salt is empty or an empty salt was passed to the authenticator config, it will quietly continue without salt. I like it better when there is a clear error. The developer should just set `'salt' => false` when the salt must be skipped after all.

Reply:
I have updated the previous comment to handle empty strings too.

Reply:
Committed, thanks.